### DISCUSSION TITLE
some suggestions about ”Add File Link“ function

### DISCUSSION THREAD

**Discussion by dragonlau0924** [ObsidianToAnki/Obsidian_to_Anki#555] [General]
Hello developer, thank you very much for sharing the Anki plugin. I have some suggestions. Could you please display the link passed into Anki in the "Add File Link"  as a file name instead of "Obsidian"? This would make it easier for me to know which note I have reviewed, which would be more practical.

**Comment by envico801** [comment]
Hi @dragonlau0924 , you can do that by modifying the `format_note_with_url` method.

You could do it in the following way:

```javascript
format_note_with_url(note, url, field) {
    let mdFilename = ""

    // We separate the string by "&file=".
    const splitUrl = url.split("&file=");

    // Check if the split resulted in two parts
    if (splitUrl.length === 2) {
        // Extract the part after "&file="
        const encodedPathWithExtension = splitUrl[1]; // Extracted part containing path with extension
        const decodedPathWithExtension = decodeURIComponent(encodedPathWithExtension);

        // Split the filename from its directory path
        const pathSegments = decodedPathWithExtension.split('/');
        const filenameWithExtension = pathSegments.pop()?.trim();

        // Check if the filename with extension is not empty
        if (filenameWithExtension) {
            // Split the filename from its extension
            const filenameSegments = filenameWithExtension.split('.');
            mdFilename = filenameSegments[0]?.trim();
        }
    }
    note.fields[field] += '<br><a href="' + url + `" class="obsidian-link">Obsidian${mdFilename ? " - " + mdFilename : ""}</a>`;
}
```

What this basically does:

1. It grabs the url from obsidian
2. Removes all unnecessary text and leaves only the filename.
3. Adds the filename after "Obsidian". e.g. "Obsidian - tests".
    ![New format for Obsidian link](https://github.com/ObsidianToAnki/Obsidian_to_Anki/assets/132226893/e7becff2-7bf6-4e72-a6c9-eb9bfb45b711)

---

If you want to try the new changes:

1.  Go to your obsidian vault (where you keep all your notes). `~/Obsidian/`
2.  Look for the plugin folder, it's usually in this location: `~/Obsidian/.obsidian/plugins/obsidian-to-anki-plugin`, if the folder does not appear you have to enable the option to show hidden files in your operating system.
3.  Open the `main.js` file and look for the word `format_note_with_url`
4.  Replace the whole function with the new version, at the end it should look something like this:
    ![new function working](https://github.com/ObsidianToAnki/Obsidian_to_Anki/assets/132226893/c383f691-1709-492d-873d-f97b25e54d83)
5.  Save and close the file
6.  Restart obsidian so the plugin reloads with the new changes.

I will also suggest the [change in a PR](https://github.com/ObsidianToAnki/Obsidian_to_Anki/pull/557), in case the maintainers want to implement it.
